code/go/0chain.net/blobbercore/handler/object_operation_handler.go

```diff
@@ -952,8 +953,22 @@ func (fsh *StorageHandler) WriteFile(ctx context.Context, r *http.Request) (*blo
 	}
 
 	allocationID := allocationObj.ID
-	fileOperation := getFileOperation(r)
-	existingFileRef := getExistingFileRef(fsh, ctx, r, allocationObj, fileOperation)
+	cmd := createFileCommand(r)
+	err = cmd.IsValidated(ctx, r, allocationObj, clientID)
+
+	if err != nil {
+		return nil, err
+	}
+
+	var existingFileRef *reference.Ref
+	switch rCmd := cmd.(type) {
+	case *UploadFileCommand:
+		existingFileRef = nil
+	case *UpdateFileCommand:
+		existingFileRef = rCmd.existingFileRef
+	case *FileCommandDelete:
+		existingFileRef = rCmd.existingFileRef
+	}
```

Comment:
The `*UpdateFileCommand` and `*FileCommandDelete` cases can be handled in one case in the following way:

```go
switch rCmd := cmd.(type) {
...
case *UpdateFileCommand, 
        *FileCommandDelete:
    existingFileRef = rcmd.existingFileRef
...
}
```

Also, please return error if the type was unknown.

Reply:
I think the code if I update it I should do something like this:
```
        case *UpdateFileCommand:
		existingFileRef = rCmd.existingFileRef
	case *FileCommandDelete:
		existingFileRef = rCmd.existingFileRef
	default:
		existingFileRef = nil
```

- We can't get the `rCmd.existingFileRef` if we don't specify exactly the type createFileCommand of the `rCmd`.
If we do something like this: 
```
 case *UpdateFileCommand, 
        *FileCommandDelete:
    existingFileRef = rcmd.existingFileRef
```
the rcmd.existingFileRef is not recognized.
- The Error is handled before we reach this step in:
```
        err = cmd.IsValidated(ctx, r, allocationObj, clientID)
	if err != nil {
		return nil, err
	}
	var existingFileRef *reference.Ref
	switch rCmd := cmd.(type) {
	case *UpdateFileCommand:
		existingFileRef = rCmd.existingFileRef
	case *FileCommandDelete:
		existingFileRef = rCmd.existingFileRef
	default:
		existingFileRef = nil
	}
```
Here we just affect to the existingFileRef for Checking if possible Collaborator.